README.md

```diff
@@ -1,282 +1,584 @@
+
 # Security Hub Visibility
 
+  
+
 ## A GitHub Action that syncs Security Hub findings with Jira issues
 
+  
+
 1. Gets all current findings from Security Hub, filtered by the `Severity` of the findings (default is `CRITICAL,HIGH,MEDIUM`)
 
+  
+
 2. Creates Jira issues for findings that do not already have a Jira issue
 
-- **To avoid creating duplicate issues**, the search criteria use the custom label configuration as well as default configuration for three identifying search labels: region, AWS account ID, and "security-hub."
+  
+
+-  **To avoid duplicate issues**, the search criteria use default settings for three primary identifiers: region, AWS account ID, and "security-hub," along with any custom label configurations provided (Note that modifications or deletions of labels may lead to duplicate issue creation). Additionally, a comparison of the Jira issue description and finding title is made against the findings list. If there is a title match, the decision is based on the Resources list. 
+- For example, if a Jira ticket T1 already exists for a finding with title T1, and a new finding appears with the same title T1 but a different Resources list, the implementation treats these as distinct, resulting in two tickets with the same title but different resources. 
+
+  
 
-- Each Security Hub Finding type (by title) is represented as a single issue, e.g. if there are three resources that have violated the 'S3.8' rule there will be a single S3.8 Jira issue created
+3. Closes existing Jira issues in the target project if their associated findings are no longer active. For findings with the same title, closure is determined based on the resources’ presence. For example, if there are two tickets for finding T2 with different resources, `r1` and `r2`, and `r1` is no longer present, then only the T2 ticket with `r1` will be closed
```

Comment:
Is there any performance lag that you noticed when scanning resources that are present on existing tickets?

Reply:
> Is there any performance lag that you noticed when scanning resources that are present on existing tickets?

I ran both versions for 3 times and took average time
For new enhancements: Avg Time is 1m34s for 36 issues - [test runs](https://github.com/Enterprise-CMCS/mac-fc-embedded/actions/runs/11862817467)
For old version: Avg Time is 1m30s for 36 issues - [test runs](https://github.com/Enterprise-CMCS/mac-fc-embedded/actions/runs/11862844136)
we can say that the time is not changed significantly